### ISSUE TITLE
Error occures when sentencepiece load model file in horovod mode

### ISSUE THREAD

**Issue by vincentge1986** [athena-team/athena#147]
**1. in Horovod mode, that is,  horovod_main.py is executed,  line 114 "self.sp.Load(spm_file)" in text_featurizer.py get below errors:**

[1,0]<stderr>:[libprotobuf FATAL external/com_google_protobuf/src/google/protobuf/stubs/common.cc:86] This program was compiled against version 3.6.1 of the Protocol Buffer runtime library, which is not compatible with the installed version (3.8.0).  Contact the program author for an update.  If you compiled the program yourself, make sure that your headers are from the same version of Protocol Buffers as your link-time library.  (Version verification failed in "/sentencepiece/src/builtin_pb/sentencepiece_model.pb.cc".)
[1,0]<stderr>:terminate called after throwing an instance of 'google::protobuf::FatalException'
[1,0]<stderr>:  what():  This program was compiled against version 3.6.1 of the Protocol Buffer runtime library, which is not compatible with the installed version (3.8.0).  Contact the program author for an update.  If you compiled the program yourself, make sure that your headers are from the same version of Protocol Buffers as your link-time library.  (Version verification failed in "/sentencepiece/src/builtin_pb/sentencepiece_model.pb.cc".)
[1,0]<stderr>:[zstc:17209] *** Process received signal ***
[1,0]<stderr>:[zstc:17209] Signal: Aborted (6)
[1,0]<stderr>:[zstc:17209] Signal code:  (-6) 

**2. However , in non-horovod mode, that is, main.py is  executed, everything seems good, run very smoothly......**

**Comment by vincentge1986** [comment]
**More info:
if comment line 114 "self.sp.Load(spm_file)" in text_featurizer.py in horovod mode,  horovod_main.py can run smoothly withou error,  get logs as below:** 

_[1,0]<stderr>:INFO:absl:global_steps: 1 learning_rate: 1.1180e-08       loss: -0.4852   Accuracy: nan   CTCAccuracy: nan
[1,0]<stderr>:INFO:absl:global_steps: 11        learning_rate: 1.2298e-07       loss: -0.4852   Accuracy: nan   CTCAccuracy: nan        sec/iter: 0.3472
[1,0]<stderr>:INFO:absl:global_steps: 21        learning_rate: 2.3479e-07       loss: -0.4852   Accuracy: nan   CTCAccuracy: nan        sec/iter: 0.3694
[1,0]<stderr>:INFO:absl:global_steps: 31        learning_rate: 3.4659e-07       loss: -0.4852   Accuracy: nan   CTCAccuracy: nan        sec/iter: 0.3859
[1,0]<stderr>:INFO:absl:global_steps: 41        learning_rate: 4.5839e-07       loss: -0.4852   Accuracy: nan   CTCAccuracy: nan        sec/iter: 0.3746
[1,0]<stderr>:INFO:absl:global_steps: 51        learning_rate: 5.7020e-07       loss: -0.4852   Accuracy: nan   CTCAccuracy: nan        sec/iter: 0.3920_

**Of course obveriously the training can't converge without line 114 "self.sp.Load(spm_file)", so i guess horovodrun cmd provide the protobuf lib version that sentencepiece can't use. Sorry that I'm not famaliar with horovod and sentencepiece, so need help to fix this issue. Thanks a lot!**

**Comment by vincentge1986** [comment]
**More info:
if comment line 114 "self.sp.Load(spm_file)" in text_featurizer.py in horovod mode, horovod_main.py can run smoothly withou error, get logs as below:**

_[1,0]:INFO:absl:global_steps: 1 learning_rate: 1.1180e-08 loss: -0.4852 Accuracy: nan CTCAccuracy: nan
[1,0]:INFO:absl:global_steps: 11 learning_rate: 1.2298e-07 loss: -0.4852 Accuracy: nan CTCAccuracy: nan sec/iter: 0.3472
[1,0]:INFO:absl:global_steps: 21 learning_rate: 2.3479e-07 loss: -0.4852 Accuracy: nan CTCAccuracy: nan sec/iter: 0.3694
[1,0]:INFO:absl:global_steps: 31 learning_rate: 3.4659e-07 loss: -0.4852 Accuracy: nan CTCAccuracy: nan sec/iter: 0.3859
[1,0]:INFO:absl:global_steps: 41 learning_rate: 4.5839e-07 loss: -0.4852 Accuracy: nan CTCAccuracy: nan sec/iter: 0.3746
[1,0]:INFO:absl:global_steps: 51 learning_rate: 5.7020e-07 loss: -0.4852 Accuracy: nan CTCAccuracy: nan sec/iter: 0.3920_

**Obveriously the training can't converge without line 114 "self.sp.Load(spm_file)", so I guess horovodrun provides a version of protobuf lib that sentencepiece can't use. Sorry for that I'm not famaliar with horovodrun and sentencepiece, so need some help to fix this issue. Thanks a lot!**

**Comment by vincentge1986** [comment]
**Howevers something is werid that if run my test.py in horovod mode as below, no error:**
_(venv_tf_v2)  horovodrun -np 4 -H localhost:4 python test.py
[1,0]<stdout>:Done: 5000
[1,1]<stdout>:Done: 5000
[1,2]<stdout>:Done: 5000
[1,3]<stdout>:Done: 5000_

**my test.py code:**
_import sentencepiece as spm
sp = spm.SentencePieceProcessor()
sp.Load("thuyg_20_unigram5000.model")
print("Done:", sp.GetPieceSize())_

**Comment by Some-random** [comment]
@cookingbear please take a look at this. Is this the error that occurs when using Horovod and sentencepiece together?

**Comment by JianweiSun007** [comment]
@vincentge1986 hi，protobuf maybe should reinstall, maybe this can help you https://github.com/BVLC/caffe/issues/5711

**Comment by vincentge1986** [comment]
> @cookingbear please take a look at this. Is this the error that occurs when using Horovod and sentencepiece together?

Correct, without horovod, everything is good, that is, "horovod_cmd" and "horovod_prefix" are commented in run.sh as below :
#horovod_cmd="horovodrun -np 1 -H localhost:1"
#horovod_prefix="horovod_"
then the training goes smoothly as expected.

**Comment by vincentge1986** [comment]
> 
> 
> @vincentge1986 hi，protobuf maybe should reinstall, maybe this can help you [BVLC/caffe#5711](https://github.com/BVLC/caffe/issues/5711)

Thanks for your reply. However it still doesn't work after reinstall protobuf:
pip uninstall protobuf
pip install protobuf
AND:
apt-get remove --purge libprotobuf-dev
apt-get install  libprotobuf-dev

**Comment by Some-random** [comment]
Keeping this issue open, this should be a bug.

**Comment by vincentge1986** [comment]
More info：
OS：Ubuntu 18.04 Server Version NOT Desttop version
libprotobuf-dev：3.6.1

**Comment by vincentge1986** [comment]
More info：
OS：Ubuntu 18.04 Server Version NOT Desktop version
libprotobuf-dev：3.6.1

**Comment by 8z8z8z** [comment]
This is a script for auto install horovod in centos and maybe helpful for you
https://github.com/8z8z8z/athena/blob/master/tools/build_horovod_centos_env.sh
